8th tutorial video review (#102)
w3c/wot-cg: Tutorials/whatiswot/8-JSON_Schema_in_Practice_3/8-JSON-Schema-3.txt

> In the previous videos of JSON Schema, we have shown the features and basic implementations of JSON Schema. In this video, we will deliver the actual validation process with libraries and examples.

Reviewer: ```suggestion
In the previous videos of JSON Schema, we have explained the features and basic implementations of JSON Schema. In this video, we will show the actual validation process with libraries and code examples.
```

Reviewer: deliver should be still changed to show

> Throughout this video, we will mainly focus on validation with the JavaScript library AJV and the Python library          jsonschema, but there are many other libraries in different programming languages and tools to validate JSON Schema. 

Reviewer: ajv pronounciation should be shorter, now it sounds like A  J  V. This happens in multiple places

Reviewer: ```suggestion
Throughout this video, we will mainly focus on validation with the JavaScript library AJV and the Python library          jsonschema, but there are many other libraries in different programming languages and tools to validate JSON using a JSON Schema. 
```

Reviewer: > ajv pronounciation should be shorter, now it sounds like A J V. This happens in multiple places

It is difficult to fix it though so we can leave as is.

> Here are some of the popular ones in different programming languages.  There are many more validators. For further information, you can check the JSON schema website.

Reviewer: For the links like the JSON Schema website, we should say: check out the json schema website via the link in the video description

Reviewer: This is still not fixed since it doesnt say video description. Also, we should say "JSON Schema Homepage" (not part of the original feedback)

This  is the basic validation flow that all libraries follow.
The library takes a schema for your JSON data and validates your data according to the schema. Outputs the validation result as a boolean 

So let's start with AJV! It is a JavaScript library.

This is the same temperature sensor example we used before. Let’s examine this in JavaScript code.

First, We are importing the AJV. 

> This is the Schema that describes the JSON. 

Reviewer: ```suggestion
This is the Schema that describes the JSON we are expecting. 
```

With this line of code, AJV takes schema as a parameter compiles it to function, and caches it.

This is the data to be validated. 

> This step actually validates the data and writes the result into a valid boolean. 

Reviewer: "This" is repeated too frequently.

```suggestion
Here we can see the validation step where the result is written into a valid boolean. 
```

> This step is for printing the result of the validation. If valid is true, then it prints "JSON Schema is valid". Otherwise, it prints the "JSON Schema is not valid." and the error message. 

Reviewer: We should change the slide here and not use "valid" as the variable name, it sounds wrong. Probably better to use `validationResult`

Reviewer: Additionally, we are not validating JSON Schema but JSON. So it is wrong to say "JSON Schema is valid". I think we should say "JSON is valid according to Schema"

It outputs that schema is valid since the data that we enter suits the predefined schema. 

> Now let's take a look at the same JSON Schema example in Python jsonschema. It is an implementation of the JSON Schema specification for Python. 

Reviewer: ```suggestion
Now let's take a look at the same validation example but with Python's jsonschema library. It is an implementation of the JSON Schema specification for Python. 
```

We will use the same temperature example that we use for AJV but for Python this time.
Let's take a look at the code in detail. 

With this line of code, we are importing the function from the library. 

This is the same schema in the previous example. 

This is the data to be validated. 

This validate function takes the data and the schema validates it and prints the result. If the data is valid, it does not print anything. Otherwise, it prints the error message. 

The code outputs that the data is not valid since the property "temperatureOf" needs to be string. However, in the given data it is an integer. 

> In the next video, we will talk about JSON and what it is detailly.

Reviewer: Next video is definitely not JSON. We have OSI Layers in the list but maybe the semantic web part will come there. To avoid changing the video, we can simply say something like below:

```suggestion
This concludes the videos on JSON Schema. You can continue with the tutorial to learn more about the Web of Things.
```
